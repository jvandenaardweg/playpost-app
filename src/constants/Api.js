export const PRODUCTION_API_URL = 'http://medium-audio.herokuapp.com'
export const DEVELOPMENT_API_URL = 'http://localhost:3000'
