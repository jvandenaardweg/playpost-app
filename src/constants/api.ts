const PRODUCTION_API_URL = 'https://api.postplay.app';
const STAGING_API_URL = 'https://api-staging.postplay.app';
const DEVELOPMENT_API_URL = 'http://localhost:3000';

interface ApiEnvironments {
  [key: string]: string;
}

const environments: ApiEnvironments = {
  production: PRODUCTION_API_URL,
  staging: STAGING_API_URL,
  development: DEVELOPMENT_API_URL
};

export const API_URL = environments[process.env.NODE_ENV || 'production'];
