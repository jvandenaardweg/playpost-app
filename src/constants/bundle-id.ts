export const APP_BUNDLE_ID = 'com.aardwegmedia.playpost';
