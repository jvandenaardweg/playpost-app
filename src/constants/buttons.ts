export const defaultHitslop = {
  top: 20,
  right: 20,
  bottom: 20,
  left: 20
};

export const smallHitslop = {
  top: 5,
  right: 5,
  bottom: 5,
  left: 5
};

export const mediumHitslop = {
  top: 10,
  right: 10,
  bottom: 10,
  left: 10
};
