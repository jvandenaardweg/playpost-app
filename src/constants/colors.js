const tintColor = '#037DE2';
const black = '#000000';
const grayDark = '#333C46';
const grayLight = '#cccccc';
const grayLightest = '#EEF0F4';
const appBackground = '#EBEBEB';
const white = '#ffffff';
const blue = tintColor;
const red = 'red';
const orange = '#EAEB5E';
const orangeDark = '#666804';

export default {
  tintColor,
  black,
  grayDark,
  grayLight,
  grayLightest,
  white,
  blue,
  red,
  orange,
  appBackground,

  tabIconDefault: grayLight,
  tabIconSelected: tintColor,
  tabBar: white,

  articleBackground: white,

  titleDefault: black,
  paragraphDefault: black,
  paragraphGrayed: grayDark,
  borderDefault: grayLightest,

  controlButtonDefault: black,
  controlButtonActive: tintColor,

  errorBackground: red,
  errorText: white,
  warningBackground: orange,
  warningText: orangeDark,
  noticeBackground: tintColor,
  noticeText: white,
};
