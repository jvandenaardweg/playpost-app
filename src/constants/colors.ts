const tintColor = '#037DE2';
const black = '#000000';
const grayDarkest = '#222222';
const grayDarker = '#444444';
const grayDark = '#888888';
const gray = '#bfbfbf';
const grayLight = '#e5e5e5';
const grayLightest = '#EEF0F4';
const appBackground = '#F1F1F1';
const white = '#ffffff';
const blue = tintColor;
const red = '#FF1017';
const redDark = '#b52424';
const orange = '#EAEB5E';
const orangeDark = '#666804';
const green = '#03A87C';

export default {
  tintColor,
  black,
  grayDarkest,
  grayDarker,
  grayDark,
  gray,
  grayLight,
  grayLightest,
  white,
  blue,
  red,
  redDark,
  orange,
  green,
  appBackground,

  tabIconDefault: gray,
  tabIconSelected: black,
  tabBar: white,

  articleBackground: white,

  titleDefault: black,
  paragraphDefault: grayDarkest,
  paragraphGrayed: grayDark,
  borderDefault: grayLight,

  controlButtonDefault: black,
  controlButtonActive: tintColor,

  errorBackground: red,
  errorText: white,
  warningBackground: orange,
  warningText: orangeDark,
  noticeBackground: tintColor,
  noticeText: white,
};
