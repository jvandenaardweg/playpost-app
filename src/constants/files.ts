import RNFS from 'react-native-fs';

export const LOCAL_STORAGE_PATH = `${RNFS.DocumentDirectoryPath}/audiofiles`;
