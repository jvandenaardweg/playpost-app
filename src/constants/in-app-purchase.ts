import { Platform } from 'react-native';

export const SUBSCRIPTION_PRODUCT_ID_FREE = Platform.select({
  ios: 'free',
  android: 'free'
});

export const SUBSCRIPTION_PRODUCT_ID_PREMIUM = Platform.select({
  ios: 'com.aardwegmedia.playpost.premium',
  android: 'com.aardwegmedia.playpost.android.premium'
});

export const SUBSCRIPTION_PRODUCT_ID_PLUS = Platform.select({
  ios: 'com.aardwegmedia.playpost.subscription.plus',
  android: 'com.aardwegmedia.playpost.android.plus'
});

export const SUBSCRIPTION_PRODUCT_IDS = [SUBSCRIPTION_PRODUCT_ID_PREMIUM, SUBSCRIPTION_PRODUCT_ID_PLUS]
