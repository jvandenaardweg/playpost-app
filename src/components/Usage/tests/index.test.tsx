import React from 'react';
import { fireEvent, render, RenderAPI } from 'react-native-testing-library';

import { getUpgradeButtonTitle, getUpgradeMessage, Usage } from '../index';

import userMock from '../../../../tests/__mocks__/user';
import { SUBSCRIPTION_PRODUCT_ID_FREE, SUBSCRIPTION_PRODUCT_ID_PREMIUM } from '../../../constants/in-app-purchase';

const onPressUpgradeHandler = jest.fn();

const defaultProps: any = {
  user: userMock,
  activeSubscriptionProductId: SUBSCRIPTION_PRODUCT_ID_FREE,
  onPressUpgrade: onPressUpgradeHandler,
  userHasSubscribedBefore: false
}

describe('Usage', () => {
  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeEach(() => {
      const props = {
        ...defaultProps
      }

      wrapper = render(<Usage {...props} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should render the used minutes correctly', () => {
      expect(wrapper.getByTestId('Usage-Text-minutes-used').props.children).toBe('of 30 minutes used');
    });

    it('should render the used percentage correctly', () => {
      expect(wrapper.getByTestId('Usage-Text-percentage').props.children).toBe('0%');
    });

    it('should render the progress width correctly', () => {
      expect(wrapper.getByTestId('Usage-View-progress').props.style[1].width).toBe('0%');
    });

    it('should render the progress width correctly when usage is above 100%', () => {
      const props = {
        ...defaultProps,
        user: {
          ...userMock,
          used: {
            audiofiles: {
              currentMonthInSeconds: 999999
            }
          },
          available: {
            audiofiles: {
              currentMonthInSeconds: 0
            }
          },
          limits: {
            audiofiles: {
              limitSecondsPerMonth: 1800,
              limitSecondsPerArticle: 1800
            }
          }
        }
      }

      wrapper.update(<Usage {...props} />);

      expect(wrapper.getByTestId('Usage-View-progress').props.style[1].width).toBe('100%');
      expect(wrapper.getByTestId('Usage-Text-percentage').props.children).toBe('100%');
    });

    it('should render an upgrade button when on a free subscription', () => {
      expect(wrapper.queryByTestId('Usage-Button-upgrade')).toBeTruthy();
    });

    it('should render an upgrade button when on a premium subscription', () => {
      const props = {
        ...defaultProps,
        activeSubscriptionProductId: SUBSCRIPTION_PRODUCT_ID_PREMIUM
      }

      wrapper.update(<Usage {...props} />);

      expect(wrapper.queryByTestId('Usage-Button-upgrade')).toBeTruthy();
    });

    it('should render the correct button title', () => {
      expect(wrapper.getByTestId('Usage-Button-upgrade').props.title).toBe('Start free Premium or Plus trial');
    });

    it('should render the correct upgrade message', () => {
      expect(wrapper.getByTestId('Usage-Text-upgrade-message').props.children).toBe('Upgrade for more minutes and Premium quality voices.');
    });

    it('should not render an upgrade button when on a plus subscription', () => {
      const props = {
        ...defaultProps,
        activeSubscriptionProductId: 'com.aardwegmedia.subscriptions.plus'
      }

      wrapper.update(<Usage {...props} />)

      expect(wrapper.queryByTestId('Usage-Button-upgrade')).toBeFalsy();
    });
  });

  describe('interacting', () => {
    let wrapper: RenderAPI;

    beforeEach(() => {
      const props = {
        ...defaultProps
      }

      wrapper = render(<Usage {...props} />);
    });

    it('should fire onPressUpgrade when pressing the upgrade button', () => {
      fireEvent.press(wrapper.getByTestId('Usage-Button-upgrade'));
      expect(onPressUpgradeHandler).toHaveBeenCalledTimes(1);
      expect(onPressUpgradeHandler).toHaveBeenCalledWith(SUBSCRIPTION_PRODUCT_ID_PREMIUM);
    });

    it('getUpgradeMessage should return the correct upgrade message when on a premium subscription', () => {
      expect(getUpgradeMessage(SUBSCRIPTION_PRODUCT_ID_PREMIUM)).toBe('Upgrade for nearly unlimited minutes.');
    });

    it('getUpgradeMessage should return the correct upgrade message when on a free subscription', () => {
      expect(getUpgradeMessage(SUBSCRIPTION_PRODUCT_ID_FREE)).toBe('Upgrade for more minutes and Premium quality voices.');
    });

    it('getUpgradeMessage should return the correct upgrade message when on a plus subscription', () => {
      expect(getUpgradeMessage('com.aardwegmedia.playpost.subscriptions.plus')).toBe('');
    });

    it('getUpgradeButtonTitle should return the correct upgrade button title when on a free subscription when a user has subscribed before', () => {
      expect(getUpgradeButtonTitle(SUBSCRIPTION_PRODUCT_ID_FREE, true)).toBe('Upgrade to Premium or Plus');
    });

    it('getUpgradeButtonTitle should return the correct upgrade button title when on a free subscription when a user has not subscribed before', () => {
      expect(getUpgradeButtonTitle(SUBSCRIPTION_PRODUCT_ID_FREE, false)).toBe('Start free Premium or Plus trial');
    });

    it('getUpgradeButtonTitle should return the correct upgrade button title when on a premium subscription when a user has subscribed before', () => {
      expect(getUpgradeButtonTitle(SUBSCRIPTION_PRODUCT_ID_PREMIUM, true)).toBe('Upgrade to Plus');
    });

    it('getUpgradeButtonTitle should return the correct upgrade button title when on a premium subscription when a user has not subscribed before', () => {
      expect(getUpgradeButtonTitle(SUBSCRIPTION_PRODUCT_ID_PREMIUM, false)).toBe('Upgrade to Plus');
    });

    it('getUpgradeButtonTitle should return the correct upgrade button title when on a plus subscription when a user has subscribed before', () => {
      expect(getUpgradeButtonTitle('com.aardwegmedia.playpost.subscriptions.plus', true)).toBe('');
    });

    it('getUpgradeButtonTitle should return the correct upgrade button title when on a plus subscription when a user has not subscribed before', () => {
      expect(getUpgradeButtonTitle('com.aardwegmedia.playpost.subscriptions.plus', false)).toBe('');
    });

  });
});
