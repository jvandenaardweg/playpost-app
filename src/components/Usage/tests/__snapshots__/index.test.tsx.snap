// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Usage rendering should render correctly 1`] = `
<View
  style={
    Object {
      "marginTop": -12,
      "paddingLeft": 12,
      "paddingRight": 12,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#037DE2",
        "borderRadius": 0,
        "marginBottom": 10,
        "padding": 12,
        "paddingBottom": 24,
      }
    }
  >
    <View
      style={
        Object {
          "paddingBottom": 12,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "marginTop": 12,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 32,
                "fontWeight": "700",
                "letterSpacing": -0.75,
              }
            }
          >
            0
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "flexGrow": 1,
              "justifyContent": "space-between",
              "marginBottom": 4,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 15,
                  "marginLeft": 4,
                  "opacity": 0.7,
                }
              }
              testID="Usage-Text-minutes-used"
            >
              of 30 minutes used
            </Text>
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 15,
                  "fontWeight": "700",
                }
              }
              testID="Usage-Text-percentage"
            >
              0%
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.2)",
            "borderRadius": 6,
            "height": 6,
            "marginTop": 6,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 6,
                "height": 6,
              },
              Object {
                "width": "0%",
              },
            ]
          }
          testID="Usage-View-progress"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "paddingTop": 12,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="Usage-Button-upgrade"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0, 0, 0, 0.4)",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Start free Premium or Plus trial
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 13,
            "marginTop": 8,
            "opacity": 0.7,
            "textAlign": "center",
          }
        }
        testID="Usage-Text-upgrade-message"
      >
        Upgrade for more minutes and Premium quality voices.
      </Text>
    </View>
  </View>
</View>
`;
